pallets/parachain-system/src/lib.rs

```diff
@@ -1069,7 +1085,7 @@ impl<T: Config> Pallet<T> {
 		match Self::host_configuration() {
 			Some(cfg) =>
 				if message.len() > cfg.max_upward_message_size as usize {
-					return Err(MessageSendError::TooBig)
+					Self::increment_ump_fee_factor();
```

Comment:
This is the wrong place to put the increment. The condition here checks whether the individual message size is greater than the limit, and does _not_ check for the queue size.